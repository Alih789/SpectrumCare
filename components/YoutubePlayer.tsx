import { useNavigation } from '@react-navigation/native';
import React, { useCallback, useEffect, useState } from 'react'
import YoutubeIframe from 'react-native-youtube-iframe';


interface YoutubePlayerProps {
    height: number,
    width: number,
    videoId: string
}
export default function YoutubePlayer({
    height,
    width,
    videoId,
} : YoutubePlayerProps) {
    const [playing, setPlaying] = useState(false)
  const onStateChanged = useCallback( (state:string) =>{
    if(state === "ended"){
      setPlaying(false)
    }
    if(state === "playing"){
      setPlaying(true)
    }
    if(state === "paused"){
      setPlaying(false)
    }
  },[])

  const navigation = useNavigation();
  useEffect(() => {
    const blur = navigation.addListener('blur', ()=> {setPlaying(false)})
    const focus = navigation.addListener('focus', ()=> {setPlaying(true)})
    // return blur, focus
  }, [navigation])
  
  
  

  return (
    <YoutubeIframe
        play={playing}
        videoId={videoId}
        height={height}
        width={width}
        webViewStyle={{opacity: 0.99}}
        onChangeState={onStateChanged}
        initialPlayerParams={{modestbranding: true}}
      />
  )
}
