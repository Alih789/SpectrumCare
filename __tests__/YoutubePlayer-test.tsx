/** @jest-environment jsdom */

import 'react-native';
import React, {useState as useStateMock} from 'react';
import YoutubePlayer from '../components/YoutubePlayer';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}));

const mockedDispatch = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      addListener: jest.fn(),
      getParent: jest.fn(),
      dispatch: mockedDispatch,
    }),
  };
});

describe('YoutubePlayer', () => {
  const setState = jest.fn();

  beforeEach(() => {
    useStateMock.mockImplementation(init => [init, setState]);
  });

  test('render yt player', () => {
    const tree = renderer
      .create(<YoutubePlayer height={10} width={10} videoId="test" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
