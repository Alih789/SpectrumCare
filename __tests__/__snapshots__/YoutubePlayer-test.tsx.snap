// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YoutubePlayer render yt player 1`] = `
<View>
  <YoutubeIframe
    height={10}
    onChangeState={[Function]}
    play={false}
    videoId="test"
    webViewStyle={
      {
        "opacity": 0.99,
      }
    }
    width={10}
  />
</View>
`;
